backend/api/v1/database_service.go

```diff
@@ -84,13 +85,122 @@ func (s *DatabaseService) ListDatabases(ctx context.Context, request *v1pb.ListD
 }
 
 // UpdateDatabase updates a database.
-func (*DatabaseService) UpdateDatabase(_ context.Context, _ *v1pb.UpdateDatabaseRequest) (*v1pb.Database, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method UpdateDatabase not implemented")
+func (s *DatabaseService) UpdateDatabase(ctx context.Context, request *v1pb.UpdateDatabaseRequest) (*v1pb.Database, error) {
+	if request.Database == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "database must be set")
+	}
+	if request.UpdateMask == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "update_mask must be set")
+	}
+
+	environmentID, instanceID, databaseName, err := getEnvironmentInstanceDatabaseID(request.Database.Name)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	database, err := s.store.GetDatabaseV2(ctx, &store.FindDatabaseMessage{
+		EnvironmentID: &environmentID,
+		InstanceID:    &instanceID,
+		DatabaseName:  &databaseName,
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+	if database == nil {
+		return nil, status.Errorf(codes.NotFound, "database %q not found", databaseName)
+	}
+
+	patch := &store.UpdateDatabaseMessage{}
+	for _, path := range request.UpdateMask.Paths {
+		switch path {
+		case "database.project":
+			projectID, err := getProjectID(request.Database.Project)
+			if err != nil {
+				return nil, status.Errorf(codes.InvalidArgument, err.Error())
+			}
+			project, err := s.store.GetProjectV2(ctx, &store.FindProjectMessage{
+				ResourceID:  &projectID,
+				ShowDeleted: true,
+			})
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, err.Error())
+			}
+			if project == nil {
```

Comment:
Do we need to check if the project is deleted?

Reply:
Done.